returnn/dataset.py

```diff
@@ -90,3 +91,64 @@ def run(self):
 
             self.out_mean.set(total_mean)
             self.out_std_dev.set(numpy.sqrt(total_var))
+
+
+def get_returnn_length_hdfs(
+    dataset_dict: Dict[str, Any],
+    extern_data: Dict[str, Any],
+    dataset_keys: List,
+    returnn_exe: tk.Path,
+    returnn_root: tk.Path,
+    job_alias: Optional[str] = None,
+) -> Dict[str, tk.Path]:
+    """
+    Uses returnn to extract the length of sequences in the datasets after feature extraction, returns a separate hdf
+    file per dataset key
+    :param dataset_dict: RETURNN config dict for dataset
+    :param extern_data: extern data dict matching the dataset
+    :param dataset_keys: keys of the datastreams
+    :param returnn_exe:
+    :param returnn_root:
+    :param job_alias: full alias of the forward job
+    :return:
+    """
+    post_config = {"use_tensorflow": True}
+    config = {
+        "eval": dataset_dict,
+        "network": {
+            "output": {
+                "class": "length",
+                "axis": "T",
+                "from": f"data:{dataset_keys[0]}",
+            },
```

Comment:
do you really need a layer called "output" in the network, or could you just have the for loop run over all dataset_keys?

thinking more of it, this is probably tied to the eval mode not working and the forward mode always dumping the layer called "output". Maybe the returnn option `forward_output_layer` can be helpful?

Reply:
from my understanding the default output is required in forward, yes. Eval would not require this iI think. 

> forward_output_layer
I think this would make it possible to not have output as a layer in the network but we would still have "output.hdf" I think, so the benefit would be marginal.